### PR TITLE
[ENTSWM-540] add tests for JSON-B

### PR DESCRIPTION
**Pull request by mirostary**

### CODE REVIEW

File: javaee/jaxrs-jsonb/pom.xml [hunk 1]
```diff
@@ -47,6 +47,10 @@
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.spec.javax.json</groupId>
+            <artifactId>jboss-json-api_1.1_spec</artifactId>
+        </dependency>
```
Comment (Ladicek) [inline javaee/jaxrs-jsonb/pom.xml]: If this is necessary, we have a bug somewhere.

Comment (Ladicek) [reply javaee/jaxrs-jsonb/pom.xml]: Actually this is JSON-P API? This definitely shouldn't be here.

**Comment by Ladicek** [comment]
I like having tests for the `Adapter` stuff, but can you move separate them more cleanly from the existing test? Thanks.

File: javaee/jaxrs-jsonb/pom.xml (hunk 1 above)
Comment (penehyba) [reply javaee/jaxrs-jsonb/pom.xml]: It is strange that the same adapter HelloWorldAdapter.java requires additional dependency in jaxrs-jsonb (org.wildfly.swarm.ts.javaee.jaxrs.jsonb )but doesn't have this dependency in javaee (org.wildfly.swarm.ts.javaee.jsonb).

File: javaee/jaxrs-jsonb/src/main/java/org/wildfly/swarm/ts/javaee/jaxrs/jsonb/HelloWorldAdapter.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+public class HelloWorldAdapter implements JsonbAdapter<Result, Result> {
+    @Override
+    public Result adaptToJson(Result obj) {
+        return obj;
+    }
+
+    @Override
+    public Result adaptFromJson(Result obj) {
+        return obj;
+    }
```
[the remark this answers is not in the document]
Comment (penehyba) [reply javaee/jaxrs-jsonb/src/main/java/org/wildfly/swarm/ts/javaee/jaxrs/jsonb/HelloWorldAdapter.java]: I think the idea was to use the same HelloWorldAdapter in both javaee/jaxrs-jsonb and javaee/jsonb.
Unfortunately jaxrs-jsonb has no dependency to jboss-json-api_1.1_spec and javax.json.JsonObject can't be used here without necessary addition of 
<dependency>
  <groupId>org.jboss.spec.javax.json</groupId>
  <artifactId>jboss-json-api_1.1_spec</artifactId>
</dependency>
So one solution is to use two different instances of JsonbAdapter OR add dependency to jaxrs-jsonb OR not use this at all and simplify the test as it looks now. I wonder is there any extra code necessary to ensure the method adaptFromJson was invoked?

[the remark this answers is not in the document]
Comment (Ladicek) [reply javaee/jaxrs-jsonb/src/main/java/org/wildfly/swarm/ts/javaee/jaxrs/jsonb/HelloWorldAdapter.java]: I'm not sure if `jaxrs-jsonb` should also provide JSON-P, but if you want JSON-P, you should add a dependency on `jaxrs-jsonp`, not on `org.jboss.spec.javax.json:jboss-json-api_1.1_spec`.

At this point, I think the test is so much simplified it doesn't really test anything. I don't know, perhaps the existing `javaee/jsonb` tests are enough and we don't have to add more?

File: javaee/jsonb/src/main/java/org/wildfly/swarm/ts/javaee/jsonb/HelloServlet.java [hunk 3]
```diff
@@ -6,12 +6,14 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+
 import java.io.IOException;
 
-@WebServlet("/")
+@WebServlet("/hello")
```
Comment (Ladicek) [inline javaee/jsonb/src/main/java/org/wildfly/swarm/ts/javaee/jsonb/HelloServlet.java]: Why not keep this on `/` and put the `AdaptServlet` to `/adapt`, possibly using a query parameter to distinguish the `from` and `to` cases? You wouldn't need to do the `RequestDispatcher` dance.

**Comment by Ladicek** [comment]
The JAX-RS case is gone, why?

**Comment by mirostary** [comment]
JAX-RS case is gone because jaxrs-jsonb  do not support JsonObject. I will work on JAX-RS 2.1 in ENTSWM-540 and there I will consider writing JAX-RS jsonb tests.

**Comment by Ladicek** [comment]
As I mentioned above, if you need JSON-P in addition to JSON-B, which makes sense here, you should add `jaxrs-jsonp`. It would probably be best for `jaxrs-jsonb` to transitively bring in `jaxrs-jsonp` though...

**Comment by Ladicek** [comment]
Filed https://issues.jboss.org/browse/THORN-2508, fix coming shortly :-)

**Comment by Ladicek** [comment]
Here's a fix: https://github.com/thorntail/thorntail/pull/1353 Should be in Thorntail `master` branch later today, so you can wait for that.

File: javaee/jsonb/src/test/java/org/wildfly/swarm/ts/javaee/jsonb/JsonbAdapterTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        assertThat(json.getAsJsonObject().has("hello")).isTrue();
+        assertThat(json.getAsJsonObject().get("hello").getAsString()).isEqualTo("world");
+    }
+
+    @Test
+    @RunAsClient
+    public void testFromJson() throws IOException {
+        String response = Request.Get("http://localhost:8080/adapt?case=from").execute().returnContent().asString();
+        JsonElement json = new JsonParser().parse(response);
+        assertThat(json.isJsonObject()).isFalse();
```
Comment (Ladicek) [inline javaee/jsonb/src/test/java/org/wildfly/swarm/ts/javaee/jsonb/JsonbAdapterTest.java]: These two lines are useless, right?

**Comment by Ladicek** [comment]
I did a couple of finishing touches and merged, thanks!

**Pull request merged**